### DISCUSSION TITLE
PRs are created with automergeType: "branch" and stabilityDays

### DISCUSSION THREAD

**Discussion by frederickfogerty** [renovatebot/renovate#9503] [Request Help]
Hi there Renovate team! Thanks so much for the great bot and tool, it really helps us a lot 🙌 

### Description

As the title alludes to, we have `automergeType` set to `branch` and `stabilityDays` set to `3`. Our goal with `automergeType: branch` is to reduce PR noise, but unfortunately the combination of these settings seems to not work so well. 

I think that since the stability days feature is implemented as a check, Renovate sees this as a pending check as thus opens a PR, even if the tests/other checks pass. 

### Expected behaviour

A PR is only opened when the tests/other checks fail or are pending for too long

### Actual behaviour

A PR is opened if any check is pending for too long

Hope this makes sense! Our full config can be found here: https://github.com/imgix/renovate-config/blob/main/default.json

Thanks 🙏

**Comment by rarkins** [comment]
Can you confirm if the PRs are being created 25-30 hours after the branch was created? We have a setting called [prNotPendingHours](https://docs.renovatebot.com/configuration-options/#prnotpendinghours) which defaults to 25. Its purpose was so that PRs without tests or with hung pending tests don't get blocked forever, but it would be my first guess as to why PRs are being raised before stabilityDays has completed.

Can you use the timestamp of when the PR was created to find the corresponding job log? 

> I think that since the stability days feature is implemented as a check, Renovate sees this as a pending check as thus opens a PR, even if the tests/other checks pass.

Yes, it's implemented as a status check, so its pending status should prevent automerging until it's green.

**Reply by frederickfogerty** [reply]
Thanks for the quick reply @rarkins. At this stage I can't find one unfortunately! Do you think if I set `prNotPendingHours` to something like `72` or `96` (to add some buffer) that should fix the issue?

**Reply by rarkins** [reply]
I think that's definitely worth doing, I'd recommend 96 if you set stabilityDays=3.

It's kind of an independent feature, but we've tried to make it not compete or contradict the newer stabilityDays feature.

The problem was that some people configure prCreation=not-pending but then either have (a) no tests ever start, or (b) a test set a pending checkmark and never pass or fail and then a branch could get stuck without a PR forever. So setting it higher like 96 is no problem as those two problems are quite rare.

**Reply by frederickfogerty** [reply]
Thanks so much @rarkins, we'll check this out!
